nixos/modules/security/acme.xml

```diff
@@ -66,4 +66,31 @@ options for the <literal>security.acme</literal> module.</para>
 
 </section>
 
+<section><title>Using ACME certificates in Nginx</title>
+<para>In practice ACME is mostly used for retrieval &amp; renewal of certificates that will be used in a webserver like Nginx. A configuration for Nginx that uses the certificates from ACME for <literal>foo.example.com</literal> will look similar to:</para>
+
+<programlisting>
+services.nginx.httpConfig = ''
+  server {
+    server_name foo.example.com;
+    listen 443 ssl;
+    ssl_certificate     ${config.security.acme.directory}/foo.example.com/fullchain.pem;
+    ssl_certificate_key ${config.security.acme.directory}/foo.example.com/key.pem;
+    root /var/www/foo.example.com/;
+  }
+'';
+</programlisting>
+
+<para>Now Nginx will try to use the certificates that will be retrieved by ACME. However, Nginx will not start until the certificates actually exist. Additionally, ACME will not be able to retrieve certificates until Nginx can host the challenge-files for ACME over HTTP. This results in a problem where the two services need eachother to function.</para>
+
+<para>To circumvent this problem we can generate self-signed certificates that Nginx can use until ACME has retrieved new ones. This can be done manually, but the ACME module can also do this automatically with the <literal>security.acme.preliminarySelfsigned</literal> option. If we use this option we need to make sure Nginx is only started after the self-signed certificate is generated. This can be configured using:</para>
+
+<programlisting>
+security.acme.preliminarySelfsigned = true;
+systemd.services.nginx = {
+  after = [ "acme-selfsigned-certificates.target" ];
```

Comment:
For now, we could add the nginx target declaration from acme module if nginx.enable is true. Then devs can go into micro optimizations of they run apache and nginx (1% case I wouldn't care about now).